### DISCUSSION TITLE
useSWR on server-side?

### DISCUSSION THREAD

**Discussion by l-you** [vercel/next.js#39002] [Ideas]
I know there is recommended way to handle data fetching: `useSWR` hook. 
For server-side pre-rendering it suggest to use `getStaticProps` in each page.

But can it be possible to handle it automatically?
useSWR is a hook, not called conditionally. Data is fetched immediately. So theoretically, server knows which data, and how is it fetched. We pass fetcher function that returns single promise and can be handled by server.

Is there plan to replace [this example](https://swr.vercel.app/examples/ssr) with something like
`
const { data, error } = useSWR("https://api.github.com/repos/vercel/swr",{
 ssr:true
});
`
?

**Comment by buesing** [comment]
Watching [this RFC](https://github.com/acdlite/rfcs/blob/first-class-promises/text/0000-first-class-support-for-promises.md#avoiding-an-uncanny-valley-between-server-and-client) which might help with this problem.

**Comment by l-you** [comment]
In 2023 "appDir" solves this problem.

**Reply by jfeldstein** [reply]
Can you explain how?

**Reply by l-you** [reply]
@jfeldstein with [React Server Components](https://nextjs.org/docs/app/building-your-application/rendering/server-components)
